examples/dlint/js.rs

```diff
@@ -0,0 +1,249 @@
+    Ok(ModuleSpecifier::resolve_import(specifier, referrer)?)
+  }
+
+  fn load(
+    &self,
+    _op_state: Rc<RefCell<OpState>>,
+    module_specifier: &ModuleSpecifier,
+    _maybe_referrer: Option<ModuleSpecifier>,
+    _is_dynamic: bool,
+  ) -> Pin<Box<deno_core::ModuleSourceFuture>> {
+    let module_specifier = module_specifier.clone();
+    async move {
+      let path = module_specifier.as_url().to_file_path().unwrap();
+      let content = std::fs::read_to_string(path)?;
+      let module = deno_core::ModuleSource {
+        code: content,
+        module_url_specified: module_specifier.to_string(),
+        module_url_found: module_specifier.to_string(),
+      };
+      Ok(module)
+    }
+    .boxed_local()
+  }
+}
+
+fn create_dummy_source(plugin_paths: &[&str]) -> String {
+  let mut dummy_source = String::new();
+  for (i, p) in plugin_paths.iter().enumerate() {
+    dummy_source += &format!(
+      "import Plugin{number} from '{path}';\n",
+      number = i,
+      path = p
+    );
+  }
+  dummy_source += r#"Deno.core.ops();
+const programAst = Deno.core.jsonOpSync('get_program', {});
+let plugin;
+"#;
+  for plugin_number in 0..plugin_paths.len() {
+    dummy_source +=
+      &format!("plugin = new Plugin{number}();\n", number = plugin_number);
+    dummy_source += "Deno.core.jsonOpSync('add_diagnostics', plugin.collectDiagnostics(programAst));\n";
+    dummy_source +=
+      "Deno.core.jsonOpSync('add_code', { code: plugin.ruleCode() });\n";
+  }
+
+  dummy_source
+}
+
+impl Plugins for JsRuleRunner {
+  fn run(&mut self, context: &mut Context, program: Program) {
+    self.runtime.register_op(
+      "get_program",
+      deno_core::json_op_sync(
+        move |_state: &mut OpState,
+              _args: Value,
+              _bufs: &mut [ZeroCopyBuf]|
+              -> Result<Value, AnyError> {
+          Ok(serde_json::json!(program))
+        },
+      ),
+    );
+
+    let specifier = ModuleSpecifier::resolve_url_or_path("dummy.js").unwrap();
+
+    // TODO(magurotuna): `futures::executor::block_on` doesn't seem ideal, but works for now
+    let module_id = deno_core::futures::executor::block_on(
+      self
+        .runtime
+        .load_module(&specifier, Some(self.dummy_source.clone())),
+    )
+    .unwrap();
+    deno_core::futures::executor::block_on(
+      self.runtime.mod_evaluate(module_id),
+    )
+    .unwrap();
+
+    let diagnostics = self
+      .runtime
+      .op_state()
+      .borrow_mut()
+      .try_take::<Diagnostics>()
+      .unwrap_or_else(Vec::new);
+    let codes = self
+      .runtime
+      .op_state()
+      .borrow_mut()
+      .try_take::<Codes>()
+      .unwrap_or_else(HashSet::new);
+
+    diagnostics.into_iter().for_each(|d| {
+      if let Some(hint) = d.hint {
+        context.add_diagnostic_with_hint(d.span, d.code, d.message, hint);
+      } else {
+        context.add_diagnostic(d.span, d.code, d.message);
+      }
+    });
+    context.set_plugin_codes(codes);
```

Comment:
I don't like this API - we should get list of all registered rules before running them, take a look at my suggestion for `registerRule` API